### PR TITLE
Kurdish support

### PR DESCRIPTION
**Pull request by ravarage**
I was able to make full kurdish support and add selection from default-config.ini file 
I made ArabicV2 for cases like issue 22

### CODE REVIEW

**Comment by ablkvo** [comment]
Hi Thank you very much for supporting the Kurdish language. I modified the arabic reshaper package based on these changes. for a string like:

> له‌رۆژنامه‌نووسه‌ هه‌ره‌ باشه‌كانی باشووری وده‌ڤه‌ری كه‌لاره‌. ئه‌زموونێكی دووروودرێژی رۆژنامه‌وانی و رادیوێی هه‌یه‌.له‌گه‌ڵ كاكه ئازاد له‌رۆژنامه‌ی"رۆژنامه‌"پێكه‌وه‌ كردووه‌ و له‌نێزیكه‌وه‌ له‌جموجۆڵ ومێژووی وبه‌رپرسیانه‌ی هه‌یه‌.

After reshaping and wring it to ta text file I got this:

> .ﻪﯾﻪﻫ یﻪﻧﺎﯿﺳﺮﭘرﻪﺑو یووﮋﻣو ڵﯚﺟﻮﻤﺟﻪﻟ هوﻪﻜﯾﺰﻧﻪﻟ و هوودﺮﻛ هوﻪﻜﭘ"ﻪﻣﺎﻧژۆر"یﻪﻣﺎﻧژۆرﻪﻟ دازﺎﺋ ﻪﻛﺎﻛ ڵﻪﮔﻪﻟ.ﻪﯾﻪﻫ ﯽﻮﯾدار و ﯽﻧاوﻪﻣﺎﻧژۆر یﮋردووروود ﯽﻜﻧوﻮﻣزﻪﺋ .هرﻻﻪﻛ یرﻪﭬهدو یروﻮﺷﺎﺑ ﯽﻧﺎﻛﻪﺷﺎﺑ هرﻪﻫ ﻪﺳوﻮﻧﻪﻣﺎﻧژۆرﻪﻟ

And using the PIL to show it on an image I got this:

[image](http://s6.picofile.com/file/8386598034/kawe.jpg)

Any help would be really appreciated. Thank you very much.

File: arabic_reshaper/default-config.ini [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [ArabicReshaper]
-# Supported languages are: [Arabic]
+# Supported languages are: [Arabic][ArabicV2][Kurdish]
```
Comment (mpcabd) [inline arabic_reshaper/default-config.ini]:
```suggestion
# Supported languages are: [Arabic, ArabicV2, Kurdish]
```

File: arabic_reshaper/default-config.ini [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 # More languages might be supported soon.
+# Arabic is recommended is work in most of the cases and support (Arabic , Urdu and Farsi)
```
Comment (mpcabd) [inline arabic_reshaper/default-config.ini]:
```suggestion
# `Arabic` is default and recommended to work in most of the cases and supports (Arabic, Urdu and Farsi)
```

File: arabic_reshaper/default-config.ini [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
+# Arabic V2 is only to be used with certain font that you run into missing chars
```
Comment (mpcabd) [inline arabic_reshaper/default-config.ini]:
```suggestion
# `ArabicV2` is only to be used with certain font that you run into missing chars
```

File: arabic_reshaper/default-config.ini [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
+# Kurdish if you are using Kurdish Sarchia font is recommended, work with both unicode and classic Arabic-Kurdish keybouard
```
Comment (mpcabd) [inline arabic_reshaper/default-config.ini]:
```suggestion
# `Kurdish` if you are using Kurdish Sarchia font is recommended, work with both unicode and classic Arabic-Kurdish keybouard 
```

File: arabic_reshaper/letters.py [hunk 5]
```diff
@@ -23,8 +23,8 @@
 
 TATWEEL = '\u0640'
 ZWJ = '\u200D'
-
-LETTERS = {
+LETTERS = {}
+LETTERS1 = {
```
Comment (mpcabd) [inline arabic_reshaper/letters.py]:
```suggestion
LETTERS_ARABIC = {
```

File: arabic_reshaper/letters.py [hunk 6]
```diff
@@ -23,8 +23,8 @@
 
 TATWEEL = '\u0640'
 ZWJ = '\u200D'
-
-LETTERS = {
+LETTERS = {}
```
Comment (mpcabd) [inline arabic_reshaper/letters.py]:
```suggestion
```

File: arabic_reshaper/letters.py [hunk 7]
```diff
@@ -184,6 +184,330 @@
     ZWJ: (ZWJ, ZWJ, ZWJ, ZWJ),
 }
 
+LETTERS2 = {
```
Comment (mpcabd) [inline arabic_reshaper/letters.py]:
```suggestion
LETTERS_ARABIC_V2 = {
```

File: arabic_reshaper/letters.py [hunk 8]
```diff
@@ -184,6 +184,330 @@
...
+    # ARABIC LETTER YEH BARREE WITH HAMZA ABOVE
+    '\u06D3': ('\u06D3', '', '', '\uFBB1'),
+    # Kurdish letter YEAH
+    '\u06ce': ('\uE004', '\uE005', '\uE006', '\uE004'),
+    # Kurdish letter Hamza same as arabic Teh without the point
+    '\u06d5': ('\u06d5', '', '', '\uE000'),
+    # ZWJ
+    ZWJ: (ZWJ, ZWJ, ZWJ, ZWJ),
+}
+LETTERS3 = {
```
Comment (mpcabd) [inline arabic_reshaper/letters.py]:
```suggestion
LETTERS_KURDISH = {
```

**Comment by ravarage** [comment]
I am aware of your replay now, I was total beginner, but you wrote your code so beautifully I could get around it and get the result I want it, thank you, yes I agree the way you suggested it should be like that

**Comment by ravarage** [comment]
Travis is failing should I fix it on my side as the way you did and pull merge request again?

**Comment by ravarage** [comment]
I made the changes and fix the errors that came with the changes all of them are good now

**Pull request closed without merging**